concordium-consensus/src/Concordium/MultiVersion.hs

```diff
@@ -1242,6 +1242,21 @@ startBaker mvr@MultiVersionRunner{mvBaker = Just Baker{..}, ..} = do
     -- called then the baker will be stopped.
     modifyMVarMasked_ bakerThread return
   where
+    -- Perform the given action when the consensus version requires a baker thread.
+    whenBakerRequired a = do
+        required <- withWriteLockIO mvr $ do
+            versions <- readIORef mvVersions
+            return $! case Vec.last versions of
+                EVersionedConfigurationV0{} -> True
+                EVersionedConfigurationV1{} -> False
+        if required
+            then a
+            else
+                mvLog
+                    Runner
+                    LLDebug
+                    "The current consensus version does not require a baker thread, so it will not \
+                    \be started"
```

Comment:
I think this log message isn't really useful, and I think it is confusing, to anybody apart from somebody who knows the code and its history. It's a bit hard to appreciate the distinction that a baker **thread** will not be started, but that baking will still happen, just via a different mechanism.

The code is so simple here that I don't think we need to log not doing the action.